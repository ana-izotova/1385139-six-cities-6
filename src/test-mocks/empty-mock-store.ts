import {AuthorizationStatus, Cities, FetchStatus, SortType, NameSpace} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userEmptyState: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.INIT
};

const allOffersEmptyState: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers: [],
  currentSort: SortType.POPULAR,
  isDataLoaded: false,
  fetchStatus: FetchStatus.INIT,
  favoritesHaveBeenChanged: false
};

const singleOfferEmptyState: SingleOfferInitialStateTypes = {
  offer: null,
  offersNearby: [],
  comments: [],
  isOfferLoaded: false,
  fetchStatus: FetchStatus.INIT
};

const favoritesEmptyState: FavoritesInitialStateTypes = {
  favoriteCards: [],
  areFavoriteCardsLoaded: false
};

export const emptyMockStore = {
  [NameSpace.ALL_OFFERS]: allOffersEmptyState,
  [NameSpace.SINGLE_OFFER]: singleOfferEmptyState,
  [NameSpace.USER]: userEmptyState,
  [NameSpace.FAVORITES]: favoritesEmptyState
};
