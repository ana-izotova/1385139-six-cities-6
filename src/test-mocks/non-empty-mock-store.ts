import {AuthorizationStatus, Cities, FetchStatus, NameSpace, SortType} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {currentOffer, offersNearby, comments, allOffers, favoriteOffers} from "./adapted-data-mock";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userStoreAuthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.AUTH,
  login: `jondoe@mail.com`,
  userAvatar: `avatar.jpg`,
  fetchStatus: FetchStatus.DONE
};

const userStoreUnauthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.DONE
};

const singleOfferStore: SingleOfferInitialStateTypes = {
  offer: currentOffer,
  offersNearby,
  comments,
  isOfferLoaded: true,
  fetchStatus: FetchStatus.DONE,
  error: null
};

const allOffersStore: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers,
  currentSort: SortType.POPULAR,
  isDataLoaded: true,
  fetchStatus: FetchStatus.DONE,
  favoritesHaveBeenChanged: true,
  error: null
};

const favoritesStore: FavoritesInitialStateTypes = {
  favoriteCards: favoriteOffers,
  areFavoriteCardsLoaded: true
};

export const nonEmptyMockStoreUnauthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreUnauthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

export const nonEmptyMockStoreAuthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreAuthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

