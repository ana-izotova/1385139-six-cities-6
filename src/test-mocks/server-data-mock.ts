export const testOffer = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2 = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};

export const testComment = {
  "id": 1,
  "user": {
    "id": 15,
    "is_pro": false,
    "name": `Kendall`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
  },
  "rating": 3,
  "comment": `We loved it so much, the house, the veiw, the location just great.. Highly recommend :)`,
  "date": `2021-02-12T08:04:28.647Z`,
};

export const testOfferFavorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2Favorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};


