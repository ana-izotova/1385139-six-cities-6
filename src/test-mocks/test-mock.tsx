import React, {Dispatch} from 'react';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import thunk, {ThunkDispatch} from 'redux-thunk';
import {RootStateType} from "../store/root-reducer";
import {ActionTypes} from "../store/action-types";
import {redirect} from "../store/middlewares/redirect";
import {emptyMockStoreAuthorized, emptyMockStoreUnauthorized} from "./empty-mock-store";
import {nonEmptyMockStoreUnauthorized, nonEmptyMockStoreAuthorized} from "./non-empty-mock-store";
import {AppThunk} from "../store/api-actions";

interface TestProps {
  children?: any,
  pushUrl?: string,
  emptyStore: boolean,
  authorized: boolean
}

export const TestMock: React.FC<TestProps> = ({children, pushUrl, emptyStore, authorized}) => {
  jest.spyOn(redux, `useDispatch`);
  jest.spyOn(redux, `useSelector`);

  const mockStore = configureStore<RootStateType, ThunkDispatch<RootStateType, AppThunk, ActionTypes>>([thunk, redirect]);

  let mock;
  if (emptyStore) {
    mock = authorized ? emptyMockStoreAuthorized : emptyMockStoreUnauthorized;
  } else {
    mock = authorized ? nonEmptyMockStoreAuthorized : nonEmptyMockStoreUnauthorized;
  }

  const store = mockStore(mock);
  const history = createMemoryHistory();

  if (pushUrl) {
    history.push(pushUrl);
  }

  return (
    <redux.Provider store={store}>
      <Router history={history}>
        {children}
      </Router>
    </redux.Provider>
  );
};


