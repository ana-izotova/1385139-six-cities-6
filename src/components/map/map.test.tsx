import React from 'react';
import {render} from '@testing-library/react';
import {TestMock} from "../../test-mocks/test-mock";
import Map from "./map";
import {allOffers} from "../../test-mocks/adapted-data-mock";

test(`Map component should render correctly`, () => {
  const {container} = render(
      <TestMock authorized={true} emptyStore={false}>
        <Map cards={allOffers}/>
      </TestMock>
  );

  expect(container).toMatchSnapshot();
});
