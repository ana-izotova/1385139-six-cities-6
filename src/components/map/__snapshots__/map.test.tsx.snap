// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component should render correctly 1`] = `
<div>
  <section
    class="property__map map leaflet-container leaflet-grab leaflet-touch-drag"
    id="map"
    style="position: relative;"
    tabindex="0"
  >
    <div
      class="leaflet-pane leaflet-map-pane"
      style="left: 0px; top: 0px;"
    >
      <div
        class="leaflet-pane leaflet-tile-pane"
      >
        <div
          class="leaflet-layer "
          style="z-index: 1;"
        >
          <div
            class="leaflet-tile-container leaflet-zoom-animated"
            style="z-index: 18; left: 0px; top: 0px;"
          >
            <img
              alt=""
              class="leaflet-tile"
              role="presentation"
              src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
              style="width: 256px; height: 256px; left: -130px; top: -72px;"
            />
          </div>
        </div>
      </div>
      <div
        class="leaflet-pane leaflet-shadow-pane"
      />
      <div
        class="leaflet-pane leaflet-overlay-pane"
      />
      <div
        class="leaflet-pane leaflet-marker-pane"
      >
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -5px; top: 159px; z-index: 159;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 11623px; top: -17964px; z-index: -17964;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 26858px; top: -18941px; z-index: -18941;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 11786px; top: -18010px; z-index: -18010;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 14788px; top: -32213px; z-index: -32213;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 11565px; top: -17807px; z-index: -17807;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 25707px; top: -21390px; z-index: -21390;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 26794px; top: -19006px; z-index: -19006;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 44432px; top: -43837px; z-index: -43837;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 44688px; top: -43651px; z-index: -43651;"
          tabindex="0"
        />
        <img
          alt=""
          class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
          src="./img/pin.svg"
          style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: 14916px; top: -32213px; z-index: -32213;"
          tabindex="0"
        />
      </div>
      <div
        class="leaflet-pane leaflet-tooltip-pane"
      />
      <div
        class="leaflet-pane leaflet-popup-pane"
      />
    </div>
    <div
      class="leaflet-control-container"
    >
      <div
        class="leaflet-top leaflet-left"
      />
      <div
        class="leaflet-top leaflet-right"
      />
      <div
        class="leaflet-bottom leaflet-left"
      />
      <div
        class="leaflet-bottom leaflet-right"
      >
        <div
          class="leaflet-control-attribution leaflet-control"
        >
          <a
            href="https://leafletjs.com"
            title="A JS library for interactive maps"
          >
            Leaflet
          </a>
           | © 
          <a
            href="https://www.openstreetmap.org/copyright"
          >
            OpenStreetMap
          </a>
           contributors © 
          <a
            href="https://carto.com/attributions"
          >
            CARTO
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
